sdk/servicebus/Azure.Messaging.ServiceBus/src/Amqp/AmqpUsageMeter.cs

```diff
@@ -0,0 +1,48 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using Microsoft.Azure.Amqp;
+
+namespace Azure.Messaging.ServiceBus.Amqp
+{
+    internal class AmqpUsageMeter : IAmqpUsageMeter
+    {
+        private readonly ServiceBusTransportMetrics _metrics;
+
+        public AmqpUsageMeter(ServiceBusTransportMetrics metrics)
+        {
+            _metrics = metrics;
+        }
+        public void OnTransportWrite(int bufferSize, int writeSize, long queueSize, long latencyTicks)
+        {
+            // not implemented
+        }
+
+        public void OnTransportRead(int bufferSize, int readSize, int cacheHits, long latencyTicks)
+        {
+            // not implemented
+        }
+
+        public void OnRead(AmqpConnection connection, ulong frameCode, int numberOfBytes)
+        {
+            switch (frameCode)
+            {
+                case 0x00:
+                    _metrics.LastHeartBeat = DateTimeOffset.UtcNow;
```

Comment:
Are we sure that this is an atomic assignment?   `DateTimeOffset` is a struct and will, I believe, need to be boxed.   Should we do an `Interlocked.Exchange` here to be safe?

Reply:
Nullable is a struct.